### PR TITLE
Check prepaid balance

### PR DESCRIPTION
**Pull request by iteyelmp**
Verify whether enough balance has been paid when initializing the contract. https://github.com/ethstorage/storage-contracts-v1/issues/40

### CODE REVIEW

File: contracts/EthStorageContract.sol [hunk 1]
```diff
@@ -30,6 +30,7 @@ contract EthStorageContract is StorageContract, Decoder {
         address _treasury,
         address _owner
     ) public payable initializer {
+        require(msg.value >= prepaidAmount, "prepaid amount is not enough");
```
Comment (syntrust) [inline contracts/EthStorageContract.sol:33]: `prepaidAmount` is not set value yet, is it `_prepaidAmount` that we should check? 
Maybe a better place is inside `__init_storage` in `StorageContract`?

File: scripts/deploy.js [hunk 2]
```diff
@@ -52,16 +53,20 @@ async function deployContract() {
...
     1048576, // nonceLimit 1024 * 1024 = 1M samples and finish sampling in 1.3s with IO rate 6144 MB/s: 4k * 2(random checks) / 6144 = 1.3s
     treasuryAddress, // treasury
     ownerAddress,
   ]);
   console.log(impl, ownerAddress, data);
   const EthStorageUpgradeableProxy = await hre.ethers.getContractFactory("EthStorageUpgradeableProxy");
-  const ethStorageProxy = await EthStorageUpgradeableProxy.deploy(impl, ownerAddress, data, { gasPrice: gasPrice });
+  const ethStorageProxy = await EthStorageUpgradeableProxy.deploy(impl, ownerAddress, data, {
+    gasPrice: gasPrice,
+    value: prepaidAmount
```
Comment (syntrust) [inline scripts/deploy.js:65]: I'm not sure if we allocate 3145 ETH when deploying the contract. Do we?

File: scripts/deploy.js [hunk 3]
```diff
@@ -73,11 +78,11 @@ async function deployContract() {
...
-  // fund 0.5 eth into the storage contract to give reward for empty mining
-  const ethStorage = StorageContract.attach(ethStorageProxy.address);
-  const tx = await ethStorage.sendValue({ value: hre.ethers.utils.parseEther("0.5") });
-  await tx.wait();
-  console.log("balance of " + ethStorage.address, await hre.ethers.provider.getBalance(ethStorage.address));
+  // // fund 0.5 eth into the storage contract to give reward for empty mining
+  // const ethStorage = StorageContract.attach(ethStorageProxy.address);
+  // const tx = await ethStorage.sendValue({ value: hre.ethers.utils.parseEther("0.5") });
+  // await tx.wait();
+  // console.log("balance of " + ethStorageProxy.address, await hre.ethers.provider.getBalance(ethStorageProxy.address));
```
Comment (syntrust) [inline scripts/deploy.js:86]: Changes made to deploy.js may need to apply to deployL2.js.

File: scripts/deploy.js [hunk 4]
```diff
@@ -52,16 +53,20 @@ async function deployContract() {
 
   const data = implContract.interface.encodeFunctionData("initialize", [
     4718592000, // minimumDiff 5 * 3 * 3600 * 1024 * 1024 / 12 = 4718592000 for 5 replicas that can have 1M IOs in one epoch
-    3145728000000000000000n, // prepaidAmount - 50% * 2^39 / 131072 * 1500000Gwei, it also means 3145 ETH for half of the shard
+    prepaidAmount, // prepaidAmount - 50% * 2^39 / 131072 * 1500000Gwei, it also means 3145 ETH for half of the shard
```
Comment (ping-ke) [inline scripts/deploy.js:56]: remove comments? 
no need "// prepaidAmount - 50% ..."

**Comment by qzhodl** [comment]
I think we can close this PR and related issue since we are implementing [this](https://github.com/ethstorage/storage-contracts-v1/pull/110)

**Comment by syntrust** [comment]
Close as the issue resolved by https://github.com/ethstorage/storage-contracts-v1/pull/110

**Pull request closed without merging**